websites/A/AnimeHeaven/presence.ts

```diff
@@ -0,0 +1,143 @@
+
+    if (buttons) {
+      const link = (document.querySelector(
+        ".now2 > div > a"
+      ) as HTMLLinkElement)?.href;
+      presenceData.buttons = [
+        {
+          label: "Current Episode",
+          url: window.location.href
+        },
+        link
+          ? {
+              label: "Episode List",
+              url: link
+            }
+          : undefined
+      ];
+    }
+
+    if (iFrameVideo) {
+      const video = iFrameVideo;
+
+      presenceData.smallImageKey = video.paused ? "pause" : "play";
+      presenceData.smallImageText = video.paused
+        ? (await strings).pause
+        : (await strings).play;
+
+      if (!video.paused) {
+        const [startTimestamp, endTimestamp] = presence.getTimestamps(
+          video.currentTime,
+          video.duration
+        );
+        presenceData.startTimestamp = startTimestamp;
+        presenceData.endTimestamp = endTimestamp;
+      }
+
+      return !video.paused;
+    }
+  },
+  "/search"(presenceData: PresenceData) {
+    presenceData.state = "Searching...";
+    presenceData.smallImageKey = "search";
+
+    const searchParams = new URLSearchParams(window.location.search);
+    let searchQuery = searchParams.get("q");
+
+    if (searchQuery) {
+      if (searchQuery.length > 18) {
+        searchQuery = `${searchQuery.substring(0, 18)}…`;
+      }
+      presenceData.details = `"${searchQuery}"`;
+    }
+  },
+  "/detail"(presenceData: PresenceData) {
+    const title = document.querySelector(
+      ".infobox .infoboxc .infodesbox .infodes h1"
+    )?.textContent;
+
+    if (title) {
+      presenceData.state = "Viewing info...";
+      presenceData.details = title;
+    } else {
+      presenceData.state = "Viewing info for a title...";
+    }
+  },
+  "/animeheaven.eu"(presenceData: PresenceData) {
+    presenceData.state = "Viewing front page...";
+  },
+  "/anime-list"(presenceData: PresenceData) {
+    presenceData.details = "Latest Anime";
+  },
+  "/dubbed-anime"(presenceData: PresenceData) {
+    presenceData.details = "Latest Dubbed Anime";
+  },
+  "/anime-series"(presenceData: PresenceData) {
+    presenceData.details = "Anime Series";
+  },
+  "/anime-movies"(presenceData: PresenceData) {
+    presenceData.details = "Anime Movies";
+  },
+  "/ongoing"(presenceData: PresenceData) {
+    presenceData.details = "Ongoing Anime";
+  },
+  "/popular"(presenceData: PresenceData) {
+    presenceData.details = "Popular Anime";
+  },
+  "/schedule"(presenceData: PresenceData) {
+    presenceData.details = "Anime Schedule";
+  },
+  "/"(presenceData: PresenceData) {
+    presenceData.state = "Viewing landing page...";
+  }
+};
+
+presence.on("UpdateData", async () => {
+  const buttons = await presence.getSetting("buttons"),
+    presenceData: PresenceData = {
+      largeImageKey: "logo"
+    },
+    currentPath = window.location.pathname.toLowerCase();
+
+  presenceData.state = "Browsing...";
+  presenceData.startTimestamp = browsingStamp;
```

Comment:
assign it to ``presenceData`` object directly.

Reply:
Yes that is what I'm doing